client/my-sites/sharing/main.jsx

```diff
@@ -1,90 +1,95 @@
+import { connect } from 'react-redux';
+import { find, get } from 'lodash';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
-var SectionNav = require( 'components/section-nav' ),
-	NavTabs = require( 'components/section-nav/tabs' ),
-	NavItem = require( 'components/section-nav/item' ),
-	Main = require( 'components/main' ),
-	SidebarNavigation = require( 'my-sites/sidebar-navigation' ),
-	utils = require( 'lib/site/utils' ),
-	sites = require( 'lib/sites-list' )();
-
+import { canCurrentUser } from 'state/selectors';
+import DocumentHead from 'components/data/document-head';
+import Main from 'components/main';
+import NavItem from 'components/section-nav/item';
+import NavTabs from 'components/section-nav/tabs';
+import SectionNav from 'components/section-nav';
+import SidebarNavigation from 'my-sites/sidebar-navigation';
 import UpgradeNudge from 'my-sites/upgrade-nudge';
 
-module.exports = React.createClass( {
-	displayName: 'Sharing',
-
-	componentDidMount: function() {
-		debug( 'Sharing React component mounted.' );
-	},
-
-	getSelectedText: function() {
-		var selected = find( this.getFilters(), { path: this.props.path } );
-		if ( selected ) {
-			return selected.title;
-		}
-
-		return '';
-	},
-
-	getFilters: function() {
-		var site = sites.getSelectedSite(),
-			pathSuffix = sites.selected ? '/' + sites.selected : '',
-			filters = [];
+class Sharing extends Component {
+	getFilters() {
+		const { canManageOptions, site, translate } = this.props;
+		const pathSuffix = site ? '/' + site.slug : '';
+		const filters = [];
 
 		// Include Connections link if all sites are selected. Otherwise,
 		// verify that the required Jetpack module is active
 		if ( ! site || ! site.jetpack || site.isModuleActive( 'publicize' ) ) {
-			filters.push( { title: this.translate( 'Connections' ), path: '/sharing' + pathSuffix, id: 'sharing-connections' } );
+			filters.push( {
+				id: 'sharing-connections',
+				path: '/sharing' + pathSuffix,
+				title: translate( 'Connections' ),
+			} );
 		}
 
 		// Include Sharing Buttons link if a site is selected and the
 		// required Jetpack module is active
-		if ( site && utils.userCan( 'manage_options', site ) &&
+		if ( site && canManageOptions &&
 			( ! site.jetpack ||
 				( site.isModuleActive( 'sharedaddy' ) && site.versionCompare( '3.4-dev', '>=' ) )
 			)
 		) {
-			filters.push( { title: this.translate( 'Sharing Buttons' ), path: '/sharing/buttons' + pathSuffix, id: 'sharing-buttons' } );
+			filters.push( {
+				id: 'sharing-buttons',
+				path: '/sharing/buttons' + pathSuffix,
+				title: translate( 'Sharing Buttons' ),
+			} );
 		}
 
 		return filters;
-	},
+	}
+
+	render() {
+		const filters = this.getFilters();
+		const selected = find( filters, { path: this.props.path } );
 
-	render: function() {
 		return (
 			<Main className="sharing">
+				<DocumentHead title={ this.props.translate( 'Sharing', { textOnly: true } ) } />
```

Comment:
Thanks for asking that! I wondered myself but not enough to actually research it.

Before React 0.14 when `ReactElements` became plain objects, the  option was needed to override `ReactElement.prototype.toString` to have control over how the object is converted to a string. The option is unused since #2455, so I'll remove it